### PR TITLE
DT-1172

### PR DESCRIPTION
**Pull request by siren**
The PR is done - it:

- [ ] follows the style and naming rules (passes `npm run lint`)
- [ ] doesn't break anything (passes `npm run test-local` and `npm run test-browserstack`)
- [ ] design is as expected. NOTE! visuals are compared using HSL theme (`CONFIG=hsl npm run dev`).
-- If no design changes: `BS_USERNAME=user BS_ACCESS_KEY=key npm run test-visual` passes
-- If design changes: `BS_USERNAME=user BS_ACCESS_KEY=key npm run test-visual-update` to generate new images
- [ ] any changed files are transformed to ES6
- [ ] all new strings visible to users are

  * [ ] added to translations file
  * [ ] are translated to English, Finnish and Swedish (if not, add translations-needed label to PR)
  
- [ ] all changed components

  * [ ] have examples
  * [ ] are included in the style guide
  * [ ] are included in the visual tests (added to gemini tests)

If this PR fixes a bug, it includes a new test that catches the bug to prevent regressions.

### CODE REVIEW

**Comment by samuliheljo** [comment]
Let's add a feature to detect whether iOS user has allowed positioning. Then it's good to go.

File: app/routes.js [hunk 1]
```diff
@@ -349,10 +349,6 @@ const routes = (
       queries={{ content: planQueries }}
       prepareParams={preparePlanParams}
       render={{ content: SummaryPageWrapper }}
-      loadAction={params => [
```
Comment (hannesj) [inline app/routes.js]: https://github.com/HSLdevcom/digitransit-ui/blob/8f6a03aff5afa02813e680a05138df52cca1801e/app/server.js#L240-L245 Could also be removed

File: static/geolocation.js [hunk 2]
```diff
@@ -82,20 +82,25 @@ function startPositioning() {
...
+        && window && window.navigator.standalone;
+    }
     //XXX hack for windows phone & safari app mode
-    if (navigator && (
-      navigator.userAgent.indexOf('Windows Phone') !== -1 ||
-      (window && window.navigator.standalone == true && (
-        navigator.userAgent.indexOf('iPhone') !==-1) || navigator.userAgent.indexOf('iPad') !==-1)
-    )) {
+    if (isWindowsPhone() || isIDeviceApp()){
       const r = JSON.parse(typeof window !== 'undefined' && window.localStorage &&
```
Comment (hannesj) [inline static/geolocation.js]: do we need an explicit r, why not just return JSON...

File: static/geolocation.js [hunk 3]
```diff
@@ -82,15 +82,26 @@ function startPositioning() {
...
+
+    function isWindowsPhone(){
+      return navigator && navigator.userAgent.indexOf('Windows Phone') !== -1;
+    }
+
+    function isIDeviceApp() {
+      return navigator
+        && navigator.userAgent
+        && (navigator.userAgent.indexOf('iPhone') !==-1 || navigator.userAgent.indexOf('iPad') !==-1)
+        && window && window.navigator.standalone;
```
Comment (hannesj) [inline static/geolocation.js]: `&& window && window.navigator.standalone;` => `&& navigator.standalone;`

File: app/localStorageHistory.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+const POP = () => {
+  if (entries.length > 0) {
+    entries.length = entries.length - 1;
+  }
+  console.log('saving:', entries);
+  setHistory(entries);
+};
+
+export default { PUSH, POP, entries };
```
Comment (hannesj) [inline app/localStorageHistory.js]: What about `REPLACE`?

File: app/history.js [hunk 5]
```diff
@@ -1,12 +1,27 @@
...
-const isBrowser = typeof window !== 'undefined';
-
-const history = useRouterHistory((isBrowser ? createHistory : createMemoryHistory))({
+const args = {
   basename: config.APP_PATH,
-});
+};
+
+const history = isIOSApp ?
+  // custom local Storage based history for ios app mode
```
Comment (hannesj) [inline app/history.js]: Should this be moved into app/localStorageHistory.js and exported as `createLocalstorageHistory`?

File: app/component/BackButton.js [hunk 6]
```diff
@@ -15,7 +16,7 @@ export default class BackButton extends React.Component {
   // When react-tap-plugin works better
   goBack = () => {
     setTimeout(() => {
-      if (window.history.length > 1) {
+      if (hasHistoryEntries()) {
```
Comment (hannesj) [inline app/component/BackButton.js]: https://github.com/mjackson/history#properties But these were oly introduced with newer history :(

**Pull request merged**